apiserver/facades/client/sshclient/facade.go

```diff
@@ -215,3 +238,25 @@ func (facade *Facade) Proxy() (params.SSHProxyResult, error) {
 	}
 	return params.SSHProxyResult{UseProxy: config.ProxySSH()}, nil
 }
+
+// Leader is not available via the V2 Facade.
+func (a *FacadeV2) Leader(_ struct{}) {}
+
+// Leader returns the unit name of the leader for the given application.
+func (facade *Facade) Leader(entity params.Entity) (params.StringResult, error) {
+	result := params.StringResult{}
+	application, err := names.ParseApplicationTag(entity.Tag)
+	if err != nil {
+		return result, err
+	}
+	leaders, err := facade.leadershipReader.Leaders()
```

Comment:
This may return the wrong result. As the backing to this is the raft store, you could be reading stale data before quorum is reached. As I don't believe we guarantee immediate consistency, as Juju is eventually consistent, then another query to the Leader API in the future should return a non-stale leader if it is wrong.

To be sure about the consistency of a leader you would have to insert a barrier to the raft leases and watch for your write to appear before reading the leaders. We currently don't support that, but if we get bugs about this, then that is the actual fix.

Reply:
This would be an existing bug as well.  The current implementation uses the results of FullStatus, which get the leader the same way.